tests/components/acaia/test_button.py

```diff
@@ -0,0 +1,83 @@
+"""Tests for the acaia buttons."""
+
+from datetime import timedelta
+from unittest.mock import MagicMock
+
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.acaia.const import DOMAIN
+from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN, SERVICE_PRESS
+from homeassistant.const import ATTR_ENTITY_ID, STATE_UNAVAILABLE, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from tests.common import async_fire_time_changed
+
+pytestmark = pytest.mark.usefixtures("init_integration")
+
+
+BUTTONS = (
+    "tare",
+    "reset_timer",
+    "start_stop_timer",
+)
+
+
+async def test_button_presses(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    device_registry: dr.DeviceRegistry,
+    snapshot: SnapshotAssertion,
+    mock_scale: MagicMock,
+) -> None:
+    """Test the acaia buttons."""
+
+    device = device_registry.async_get_device({(DOMAIN, mock_scale.mac)})
+    assert device
+    assert device == snapshot(name="device")
```

Comment:
Snapshotting the device is not button specific as you will add more platforms in the future, so I would suggest having it in test_init